### PR TITLE
[bitnami/postgresql-ha] Fix postgresql repmgr `POSTGRES_POSTGRES_PASSWORD` generation

### LINKED ISSUE #25671: FATAL:  database "repmgr" does not exist
### Name and Version

bitnami/postgresql-ha:14.0.10

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Kubernetes version: v1.28.6
Helm: v3.13.1

### Are you using any custom parameters or values?
```
--set global.postgresql.username=user1 \
--set global.postgresql.database=db1 \
--set postgresql.audit.logConnections=true \
--set postgresql.audit.logDisconnections=true \
--set pgpool.logConnections=true \
--set pgpool.logPerNodeStatement=true \
--set witness.create=true \
--set witness.audit.logConnections=true \
--set witness.audit.logDisconnections=true \
--set metrics.enabled=true \
--set metrics.serviceMonitor.enabled=true \
--set persistence.enabled=false
```
### What do you see instead?
```
postgresql-repmgr 11:03:31.32 INFO  ==> 
postgresql-repmgr 11:03:31.33 INFO  ==> Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 11:03:31.33 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql-repmgr 11:03:31.33 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
postgresql-repmgr 11:03:31.34 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
postgresql-repmgr 11:03:31.34 INFO  ==> 
postgresql-repmgr 11:03:31.43 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 11:03:31.54 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 11:03:31.55 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 11:03:31.56 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 11:03:31.82 INFO  ==> There are no nodes with primary role. Assuming the primary role...
postgresql-repmgr 11:03:31.84 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 11:03:31.84 INFO  ==> postgresql.conf file not detected. Generating it...
postgresql-repmgr 11:03:32.25 INFO  ==> Preparing repmgr configuration...
postgresql-repmgr 11:03:32.34 INFO  ==> Initializing Repmgr...
postgresql-repmgr 11:03:32.34 INFO  ==> Initializing PostgreSQL database...
postgresql-repmgr 11:03:32.35 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql-repmgr 11:03:32.42 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
postgresql-repmgr 11:03:32.52 INFO  ==> Deploying PostgreSQL with persisted data...
postgresql-repmgr 11:03:32.63 INFO  ==> Configuring replication parameters
postgresql-repmgr 11:03:32.75 INFO  ==> Configuring fsync

postgresql-repmgr 11:03:32.83 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **
postgresql-repmgr 11:03:32.94 INFO  ==> Starting PostgreSQL in background...
waiting for server to start....2024-05-10 11:03:33.062 GMT [153] LOG:  pgaudit extension initialized
2024-05-10 11:03:33.128 GMT [153] LOG:  redirecting log output to logging collector process
2024-05-10 11:03:33.128 GMT [153] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
2024-05-10 11:03:33.128 GMT [153] LOG:  starting PostgreSQL 16.2 on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2024-05-10 11:03:33.129 GMT [153] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2024-05-10 11:03:33.129 GMT [153] LOG:  listening on IPv6 address "::", port 5432
2024-05-10 11:03:33.141 GMT [153] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2024-05-10 11:03:33.160 GMT [157] LOG:  database system was shut down at 2024-05-10 11:03:30 GMT
2024-05-10 11:03:33.172 GMT [153] LOG:  database system is ready to accept connections
 done
server started
2024-05-10 11:03:33.243 GMT [163] LOG:  connection received: host=[local]
postgresql-repmgr 11:03:33.24 INFO  ==> ** Starting repmgrd **
[2024-05-10 11:03:33] [NOTICE] repmgrd (repmgrd 5.4.1) starting up
2024-05-10 11:03:33.264 GMT [166] LOG:  connection received: host=postgresql-ha-postgresql-0.postgresql-ha-postgresql-headless.default.svc.cluster.local port=32870
2024-05-10 11:03:33.340 GMT [166] LOG:  connection authenticated: identity="repmgr" method=md5 (/opt/bitnami/postgresql/conf/pg_hba.conf:1)
2024-05-10 11:03:33.340 GMT [166] LOG:  connection authorized: user=repmgr database=repmgr application_name=repmgr
2024-05-10 11:03:33.340 GMT [166] FATAL:  database "repmgr" does not exist
[2024-05-10 11:03:33] [ERROR] connection to database failed
[2024-05-10 11:03:33] [DETAIL] 
connection to server at "postgresql-ha-postgresql-0.postgresql-ha-postgresql-headless.default.svc.cluster.local" (10.233.96.193), port 5432 failed: FATAL:  database "repmgr" does not exist

[2024-05-10 11:03:33] [DETAIL] attempted to connect using:
  user=repmgr password=Iqo2VU54uk connect_timeout=5 dbname=repmgr host=postgresql-ha-postgresql-0.postgresql-ha-postgresql-headless.default.svc.cluster.local port=5432 fallback_application_name=repmgr options=-csearch_path=
```
### Additional information

If I remove these parameters everything starts working
- global.postgresql.username=user1
- global.postgresql.database=db1

### PR DESCRIPTION
**Pull request by rafariossaa**
### Description of the change

This bumps the image used, that include some fixes for the admin user/password handling, and fixes when to render `POSTGRES_POSTGRES_PASSWORD`.

### Applicable issues

- fixes #25671

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/postgresql-ha/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # Changelog
 
-## 14.2.4 (2024-06-11)
+## 14.2.5 (2024-06-11)
 
-* [bitnami/postgresql-ha] Release 14.2.4 ([#27094](https://github.com/bitnami/charts/pull/27094))
+* Draft: [bitnami/postgresql-ha] Fix postgresql repmgr  generation ([#27095](https://github.com/bitnami/charts/pull/27095))
```
Comment (juan131) [inline bitnami/postgresql-ha/CHANGELOG.md:5]: This must be updated (contains the "draf" reference)

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 2]
```diff
@@ -189,7 +189,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) }}
+            {{- if and (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPostgresPassword" .)) (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) }}
```
Comment (juan131) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:192]: I don't understand the change `postgresql-ha.postgresqlPostgresPassword` doesn't return a boolean but the actual password which always has a value (provided or randomly created)

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/postgresql-ha/CHANGELOG.md (hunk 1 above)
Comment (juan131) [reply bitnami/postgresql-ha/CHANGELOG.md]: done at https://github.com/bitnami/charts/pull/27095/commits/0b81e5abbbc67436a019b883c8945fd5ea4f6515

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml (hunk 2 above)
Comment (rafariossaa) [reply bitnami/postgresql-ha/templates/postgresql/statefulset.yaml]: I have updated the condition.
The `POSTGRES_POSTGRES_PASSWORD` env var should be generated when:
 - A secret is used -> use the provided
 - It is provided -> use the provided
 - When the user is not `postgres` -> generate a random or use the provided.

**Review by juan131 — APPROVED** [review]

**Review by juan131 — APPROVED** [review]

**Pull request merged**